src/sub/payments.py

```diff
@@ -393,52 +397,21 @@ def create_return_data(subscriptions) -> JsonDict:
             product = retrieve_stripe_product(subscription["plan"]["product"])
         products[subscription["plan"]["product"]] = product
 
-        plan_name = format_plan_nickname(
-            product_name=product["name"], plan_interval=subscription["plan"]["interval"]
-        )
         if subscription.get("status", None) == "incomplete":
             invoice = retrieve_stripe_invoice(subscription["latest_invoice"])
             if invoice["charge"]:
                 intents = retrieve_stripe_charge(invoice["charge"])
                 logger.debug("intents", intents=intents)
                 return_data["subscriptions"].append(
-                    {
-                        "current_period_end": subscription["current_period_end"],
-                        "current_period_start": subscription["current_period_start"],
-                        "ended_at": subscription["ended_at"],
-                        "plan_name": plan_name,
-                        "plan_id": subscription["plan"]["id"],
-                        "status": subscription["status"],
-                        "subscription_id": subscription["id"],
-                        "cancel_at_period_end": subscription["cancel_at_period_end"],
-                        "failure_code": intents["failure_code"],
-                        "failure_message": intents["failure_message"],
-                    }
+                    format_subscription(subscription, product, intents)
                 )
                 continue
```

Comment:
continue is not required and can have undesired effects.

Reply:
updated